### PR TITLE
analytics component

### LINKED ISSUE #727: analytics component
by the end of this year i intend to start writing an integration for solr's [analytics component](https://lucene.apache.org/solr/guide/7_2/analytics.html).

ideally this component gets integrated into solarium, but as the analytics request handler is not enabled by default, i wonder whether that's the way to go or that writing an extension would make more sense.

the way i'm curretly looking at it, is that it will be a rather straightforward component which allows  setting functions, expressions and groupings. 

seperately i'm considerig wiriting a query builder to ease the writing of those functions, expresions and groupigs. 

what are your thoughts on this one, any suggestions or concerns?

### PR DESCRIPTION
**Pull request by mkalkbrenner**
implementation of the solr analytics component for solarium.
@ see https://lucene.apache.org/solr/guide/8_3/analytics.html

fixes #727

### CODE REVIEW

**Comment by mkalkbrenner** [comment]
@wickedOne I meanwhile merged a lot of the PRs for 6.0. Could you resolve the conficts and update this PR?

**Comment by wickedOne** [comment]
done...

**Comment by mkalkbrenner** [comment]
There was 1 failure:

1) Solarium\Tests\Core\Client\Adapter\Psr18AdapterTest::testExecuteBasicGetRequest

Expectation failed for method name is "sendRequest" when invoked 1 time(s)

Failed asserting that two arrays are identical.

--- Expected

+++ Actual

@@ @@

         0 => '127.0.0.1:8983'

     )

     'X-Request-Header' => Array &2 (

-        0 => 'some value'

-        1 => 'and another one'

+        0 => 'and another one'

     )

     'Content-Type' => Array &3 (

         0 => 'application/x-www-form-urlencoded; charset=utf-8'

     )

 )

/home/travis/build/solariumphp/solarium/src/Core/Client/Adapter/Psr18Adapter.php:46

/home/travis/build/solariumphp/solarium/tests/Core/Client/Adapter/Psr18AdapterTest.php:45

**Comment by wickedOne** [comment]
you're sure that's related to this pr?
will have a look at it this evenning...

**Comment by wickedOne** [comment]
should be fixed @ https://github.com/solariumphp/solarium/pull/731/commits/37c3cace250bf15b25569ac52e1ce5c8e2605a0f

**Comment by mkalkbrenner** [comment]
```
  Line   tests/Component/Analytics/Facet/AnalyticsFacetTest.php              
 ------ -------------------------------------------------------------------- 
  42     You should use assertNull() instead of assertSame(null, $actual).   
  72     You should use assertTrue() instead of assertSame() when expecting  
         "true"                                                              
  130    You should use assertNull() instead of assertSame(null, $actual).   
  156    You should use assertNull() instead of assertSame(null, $actual).   
 ------ -------------------------------------------------------------------- 
```

Would it be a big effort to create at least a small integration test?

**Comment by wickedOne** [comment]
i think it will actually be quite a bit of work as, as far as i know, no optional components are part of the integration tests so it requires the addition of a workflow modifying the schema.xml and reloading the core.

nevertheless i think having an integration test for this component (or any) will benefit us in the long run so i'll have a look at it in the very near future

**Comment by wickedOne** [comment]
@mkalkbrenner it appears to be impossible to include this component in the integration tests as it requires the addition of a library directive to the solrconfig.xml
```xml
<lib dir="${solr.install.dir:../../../..}/dist/" regex="solr-analytics-\d.*\.jar" />
```
and, as far as i can see, the solr api does not support adding those directives through their config api...

**Comment by wickedOne** [comment]
is there anything i can do to get this one merged?

File: CHANGELOG.md [hunk 1]
```diff
@@ -46,6 +46,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Raw XML from file in update query
 - Set input encoding for select and update queries
 - Create and configure Managed Resources
+- Addition of the analytic component
```
Comment (mkalkbrenner) [inline CHANGELOG.md:49]: This needs to be moved ;-)

**Pull request closed without merging**